packages/sidecar/src/messageBuffer.ts

```diff
@@ -60,13 +60,26 @@ export class MessageBuffer<T> {
     setTimeout(() => this.stream(readerId, atReadPos), 500);
   }
 
+  /**
+   * hard reset: drops items and subscribers and resets cursors.
+   */
   clear(): void {
     this.items = new Array(this.size);
     this.writePos = 0;
     this.head = 0;
     this.readers = new Map<string, (item: T) => void>();
   }
 
+  /**
+   * soft reset: clears buffered items but preserves subscribers
+   * and write cursor. Readers will fast-forward to the new head and continue
+   * streaming from the next write.
+   */
+  reset(): void {
+    this.items = new Array(this.size);
```

Comment:
I'd say move the code in `clear()` here (including the `writePos` and `head` reset), them make `clear()` call `this.reset()` and remove the subscribers.